### ISSUE TITLE
Leading comments omitted for Golang generation

### ISSUE THREAD

**Issue by danthegoodman1** [glideapps/quicktype#2222]
`"version": "23.0.17"`

```ts
import {
	quicktype,
	InputData,
	JSONSchemaInput,
	FetchingJSONSchemaStore
} from "quicktype-core"
// ...
return await quicktype({
    inputData,
    lang: targetLanguage,
    leadingComments: ["THIS FILE HAS BEEN GENERATED, DO NOT EDIT", JSON.stringify({updated: new Date()} as CommentMeta)],
    rendererOptions: {
      "just-types": true
    } as any
  });
```

For `go` as the `targetLanguage` I get:

```go
// Moderation queue item approved
type IaModApproved struct {
	Channel           string                 `json:"channel"`
	Context           Context                `json:"context"`
	Event             string                 `json:"event"`
	Integrations      map[string]interface{} `json:"integrations"`
	MessageID         string                 `json:"messageId"`
	OriginalTimestamp string                 `json:"originalTimestamp"`
	ProjectID         string                 `json:"projectId"`
	Properties        Properties             `json:"properties"`
	ReceivedAt        string                 `json:"receivedAt"`
	SentAt            string                 `json:"sentAt"`
	Timestamp         string                 `json:"timestamp"`
	Type              string                 `json:"type"`
	UserID            string                 `json:"userId"`
	Version           float64                `json:"version"`
}

type Context struct {
	Library *Library `json:"library,omitempty"`
}

type Library struct {
	Name    *string `json:"name,omitempty"`
	Version *string `json:"version,omitempty"`
}

type Properties struct {
	Input      *Input  `json:"Input,omitempty"`
	WorkflowID *string `json:"workflowID,omitempty"`
}

type Input struct {
	BuyerVars         *BuyerVars `json:"BuyerVars,omitempty"`
	CreatorID         *string    `json:"CreatorID,omitempty"`
	ForceModerationOn *bool      `json:"ForceModerationOn,omitempty"`
	InteractionID     *string    `json:"InteractionID,omitempty"`
}

type BuyerVars struct {
	Action  *Action  `json:"action,omitempty"`
	Default *Default `json:"default,omitempty"`
}

type Action struct {
	Message *string `json:"Message,omitempty"`
	Prompt  *string `json:"Prompt,omitempty"`
}

type Default struct {
	DisplayName *string `json:"DisplayName,omitempty"`
}
```

But with `typescript` I get:

```ts
// THIS FILE HAS BEEN GENERATED, DO NOT EDIT
// {"updated":"2023-03-20T22:40:31.241Z"}

/**
 * Moderation queue item approved
 */
export interface IaModApproved {
    channel:           string;
    context:           Context;
    event:             string;
    integrations:      { [key: string]: any };
    messageId:         string;
    originalTimestamp: string;
    projectId:         string;
    properties:        Properties;
    receivedAt:        string;
    sentAt:            string;
    timestamp:         string;
    type:              string;
    userId:            string;
    version:           number;
    [property: string]: any;
}

export interface Context {
    library?: Library;
    [property: string]: any;
}

export interface Library {
    name?:    string;
    version?: string;
    [property: string]: any;
}

export interface Properties {
    Input?:      Input;
    workflowID?: string;
    [property: string]: any;
}

export interface Input {
    BuyerVars?:         BuyerVars;
    CreatorID?:         string;
    ForceModerationOn?: boolean;
    InteractionID?:     string;
    [property: string]: any;
}

export interface BuyerVars {
    action?:  Action;
    default?: Default;
    [property: string]: any;
}

export interface Action {
    Message?: string;
    Prompt?:  string;
    [property: string]: any;
}

export interface Default {
    DisplayName?: string;
    [property: string]: any;
}
```

Note that `just-types` did not make an impact on the comments.

**Comment by inferrinizzard** [comment]
This can be done with #2482 with the `leadingComments` param

**Comment by schani** [comment]
**Triage sweep note (assisted by Claude):** confirmed reproducible on current master (v23.3.0, `68ecb284`). Reopening for context, not asking to reopen the issue itself — flagging that the fix believed to have landed (PR #2482, `leadingComments`) doesn't actually work for Go.

## Root cause

The Go renderer silently drops the `leadingComments` API option that other renderers (e.g. TypeScript) honor. In `packages/quicktype-core/src/language/Golang/GolangRenderer.ts`, the checks at lines 198 and 612 test `this.leadingComments === undefined` only to decide whether to *suppress* the default "Code generated..." header — but the renderer never actually calls `this.emitComments(this.leadingComments)` to emit the supplied comments, unlike e.g. `JavaScriptRenderer.ts:549-550`. Net effect: passing `leadingComments` to the Go target suppresses the default header and emits nothing in its place, with or without `just-types`.

Verified directly against quicktype-core: passing `leadingComments: [...]` produces the comments at the top of TypeScript output but nothing for Go.

## Assessment

- **Severity: medium** — wrong/missing output, but it's an API-only option (not exposed by the CLI flags AFAIK) and doesn't crash.
- **Difficulty: easy** — add an `emitComments(this.leadingComments)` call in Go's `emitSourceStructure`/equivalent top-of-file emission, mirroring how JavaScript does it.

## Reproduction

```js
const { quicktype, InputData, jsonInputForTargetLanguage } = require("quicktype-core");

async function run(lang, justTypes) {
    const jsonInput = jsonInputForTargetLanguage(lang);
    await jsonInput.addSource({
        name: "IaModApproved",
        samples: [JSON.stringify({ channel: "web", event: "approved", version: 1.5 })],
    });
    const inputData = new InputData();
    inputData.addInput(jsonInput);
    const result = await quicktype({
        inputData,
        lang,
        leadingComments: ["THIS FILE HAS BEEN GENERATED, DO NOT EDIT"],
        rendererOptions: justTypes ? { "just-types": true } : {},
    });
    console.log(`===== ${lang} just-types=${justTypes} =====`);
    console.log(result.lines.slice(0, 8).join("\n"));
}

(async () => {
    await run("go", true);
    await run("go", false);
    await run("typescript", true); // for comparison — this one does show the comment
})();
```

Go output (both `just-types` settings) has no trace of the comment; TypeScript output does.
